blog/_src/posts/2017-06-06-spring-2017-pl-junior-retrospective.md

```diff
@@ -0,0 +1,162 @@
+Sketch-n-Sketch is a cool system. It demonstrates that a narrow
+application of synthesis - trying to fill in the constant values in a
+program (sketching) - can be used for great effect. We were left
+wondering, however, if it was too narrow an application of synthesis
+to give much of an indication of what the entire field is like.
+
+We concluded our program synthesis segment with
+[Type-and-Example-Directed Program Synthesis][typeandexample] by Osera
+and Zdancewic, another relatively recent paper. This seems like a
+relevant paper because we are under the impression that using examples
+to do synthesis is a big thing right now. Using types to constrain the
+search is another interesting perspective on techniques for synthesis.
+
+While each of theses papers had merits, none was so comprehensive as
+to be a necessary inclusion in any future look at program synthesis
+for pl junior
+
+## Sequent Calculus
+
+We followed up the program synthesis unit with a week on the sequent
+calculus. The seminar presentation was based on a paper by
+[Herbelin][Herbelin]. [Gabriel’s thesis][gab] (chapter 4) includes
+maybe a more suitable modern introduction to the sequent calculus.
+
+It might have been better to do sequent calculus first because there
+is a modern branch of proof search based on the sequent calculus.
+Presenting this first would have allowed us to look into proof search
+for program synthesis.
+
+An additional problem is that it was insufficiently motivated. Either
+skipping the topic or spending more time on it would be preferable,
+since one week was just enough to describe the sequent calculus but
+not enough to apply it.
+
+## Logic Programming
+
+The topic was presented over two weeks. The first session
+presented/demoed Prolog as a language, and we got a sense of what
+logic programming could do. But it was a whirlwind tour, and we were
+left wondering about specific details (how proof search runs, what
+`cut` does).
+
+The second session presented the paper
+[The Semantics of Predicate Logic as a Programming Language][predlogic].
+It was interesting and insightful but left wondering how it relates to
+the implementation of real logic programming languages.
+
+In hindsight this was about as far as we could have gotten in just two
+weeks.
+
+## Bonus Rounds
+
+We also used a few weeks to read and discuss specific papers as a
+group.
+
+The first paper we read was Cardelli's [Typeful Programming][typeful].
+We picked typeful programming because Matthias has mentioned on
+occasion how important he thinks it is.
+
+It was an interesting read; more of an essay than a paper. It really
+stood out as different from the other academic publications that we
+have looked at. It’s a walk through of a language design motivating
+each design decision in practical terms, as in things that actually
+help the programmer.
+
+Cardelli thinks polymorphism (subtyping in addition to parametric) is
```

Comment:
Maybe change to "Cardelli places great importance on polymorphism (subtyping in addition to parametric), as well as..." The current version sounds awkward to me.